### DISCUSSION TITLE
ا How to build from source to the Android platform

### DISCUSSION THREAD

**Discussion by WOLFxxxxxx** [bjornbytes/lovr#784] [General]
ا
How to build from source to the Android platform

%%%%%%%%%%
It has Windows 10
And I can't get Linux 

thank you for help

**Comment by bjornbytes** [comment]
Can you install WSL?  It's kind of like a Linux VM integrated into Windows.  I use WSL to compile APKs on Windows pretty often and can definitely recommend it or help out with any issues.

**Reply by WOLFxxxxxx** [reply]
ا
I'm having problems installing wsl
Can I compile by vscode without wsl

**Reply by bjornbytes** [reply]
I haven't seen anyone do it successfully yet, but in theory it is possible.  The last time someone tried, they ran into [this issue with LuaJIT cross compilation](https://github.com/WohlSoft/LuaJIT/issues/22), but it appears to have since been fixed.  I'm trying it now to see if it works.

**Reply by WOLFxxxxxx** [reply]
thank you for help💜

**Reply by bjornbytes** [reply]
Ok I just tried compiling an APK from Windows and it works now, amazing!!  You should be able to follow the instructions in the [Compiling Guide](https://lovr.org/docs/dev/Compiling#android) with a couple windows-specific tweaks:

- Make sure you add `-G Ninja` to CMake to use the ninja generator and make sure your path includes `/path/to/android/cmake/bin` so CMake can find ninja.
- Make sure you have `JAVA_HOME` set.
- NDK 21 will not work because Jolt uses some newer clang flags.  I upgraded to 26.3.11579264 and the issue was fixed.
- I used Git Bash...this may or may not be important but I thought I'd mention it.
